osdk/tools/Dockerfile.ubuntu22.04

```diff
@@ -0,0 +1,40 @@
+# SPDX-License-Identifier: MPL-2.0
+
+# This image is for the OSDK GitHub CI. 
+# The environment is consistent with the one 
+# described in the OSDK User Guide section of the Asterinas Book.
```

Comment:
What are the pros and cons of having a separate Docker image for OSDK? I mean. Yes, OSDK is an relatively-independent project. So it makes sense that OSDK have a dedicated Docker image. But if the purpose of such a Docker image is only meant for CI, then the existing Asterinas Docker image is sufficient to run the CI, right? Having two Docker images brings more maintenance costs. So I am not entirely that pros of having a separate OSDK Docker image outweighs the cons.

Reply:
> What are the pros and cons of having a separate Docker image for OSDK?

This is because the content of Asterinas Docker image is different from that of OSDK image. The main difference is that QEMU, grub and OVMF of OSDK are installed by apt, while those of Asterinas are built from source. 

So the OSDK image is used to test whether OSDK can run properly with tools installed by apt. 

If Asterinas can also be run with the tools installed by apt, I think we can have a unified image, just build Asterinas image besed on OSDK image. But I'm not sure whether is works properly now.

@junyang-zh, How do you like replacing the QEMU, OVMF and grub in Asterinas image as installed by apt? Are there any difficulties at the moment? If not, I think we do not need to have a seperate image and I'm happy to include the change in this PR.